File: screenshot-script/capture.py
import os
import time
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager
from webdriver_manager.microsoft import EdgeChromiumDriverManager
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

SCREENSHOT_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'UI Screenshots')
if not os.path.exists(SCREENSHOT_DIR):
    os.makedirs(SCREENSHOT_DIR)

def main():
    print("Initializing WebDriver...")
    try:
        # Try Chrome first
        service = Service(ChromeDriverManager().install())
        options = webdriver.ChromeOptions()
        options.add_argument('--headless')
        options.add_argument('--window-size=1280,800')
        driver = webdriver.Chrome(service=service, options=options)
    except Exception as e:
        print(f"Chrome failed: {e}. Trying Edge...")
        # Fallback to Edge
        service = Service(EdgeChromiumDriverManager().install())
        options = webdriver.EdgeOptions()
        options.add_argument('--headless')
        options.add_argument('--window-size=1280,800')
        driver = webdriver.Edge(service=service, options=options)

    # Login
    print("Navigating to login page...")
    driver.get("http://localhost:3000/login")
    time.sleep(2)
    
    print("Entering credentials...")
    # Find email input
    email_input = driver.find_element(By.CSS_SELECTOR, 'input[type="email"]')
    email_input.send_keys("admin@saiiti.edu.in")
    
    # Find password input
    password_input = driver.find_element(By.CSS_SELECTOR, 'input[type="password"]')
    password_input.send_keys("Admin@123")
    
    # Press Enter to submit
    password_input.send_keys(Keys.RETURN)
    
    # Wait for dashboard to load
    print("Waiting for login to complete...")
    time.sleep(4)
    
    pages = [
        {"name": "Dashboard", "url": "http://localhost:3000/dashboard"},
        {"name": "Students", "url": "http://localhost:3000/students"},
        {"name": "Fee Structures", "url": "http://localhost:3000/fee-structures"},
        {"name": "Payments", "url": "http://localhost:3000/payments"},
        {"name": "Receipts", "url": "http://localhost:3000/receipts"},
        {"name": "Reports", "url": "http://localhost:3000/reports"}
    ]
    
    for page in pages:
        print(f"Capturing {page['name']}...")
        driver.get(page['url'])
        time.sleep(2) # adjust if pages need more time to load data
        
        file_path = os.path.join(SCREENSHOT_DIR, f"{page['name']}.png")
        driver.save_screenshot(file_path)
        print(f"Saved: {file_path}")
        
    print("Capturing Login separately...")
    driver.delete_all_cookies()
    driver.get("http://localhost:3000/login")
    time.sleep(2)
    login_path = os.path.join(SCREENSHOT_DIR, "Login.png")
    driver.save_screenshot(login_path)
    print(f"Saved: {login_path}")
    
    print("Done!")
    driver.quit()

if __name__ == "__main__":
    main()

File: screenshot-script/capture.js
const puppeteer = require('puppeteer');
const path = require('path');
const fs = require('fs');

const SCREENSHOT_DIR = path.join(__dirname, '..', 'UI Screenshots');

if (!fs.existsSync(SCREENSHOT_DIR)) {
    fs.mkdirSync(SCREENSHOT_DIR, { recursive: true });
}

async function captureScreenshots() {
    console.log('Launching browser...');
    const browser = await puppeteer.launch({
        headless: 'new', // or true
        args: ['--no-sandbox', '--disable-setuid-sandbox']
    });
    const page = await browser.newPage();

    // Set viewport
    await page.setViewport({ width: 1280, height: 800 });

    console.log('Navigating to login...');
    await page.goto('http://localhost:3000/login', { waitUntil: 'networkidle0' });

    console.log('Logging in...');
    await page.type('input[type="email"]', 'admin@saiiti.edu.in');

    // The password field might just be input[type="password"]
    await page.type('input[type="password"]', 'Admin@123');

    // Click the submit button (assuming it's a button with type submit or the only button)
    // await page.click('button[type="submit"]');
    // Or press Enter
    await Promise.all([
        page.keyboard.press('Enter'),
        page.waitForNavigation({ waitUntil: 'networkidle0' }),
    ]);

    const pagesToCapture = [
        { name: 'Dashboard', url: 'http://localhost:3000/dashboard' },
        { name: 'Students', url: 'http://localhost:3000/students' },
        { name: 'Fee Structures', url: 'http://localhost:3000/fee-structures' },
        { name: 'Payments', url: 'http://localhost:3000/payments' },
        { name: 'Receipts', url: 'http://localhost:3000/receipts' },
        { name: 'Reports', url: 'http://localhost:3000/reports' }
    ];

    for (const p of pagesToCapture) {
        console.log(`Capturing ${p.name}...`);
        await page.goto(p.url, { waitUntil: 'networkidle0' });
        // wait extra 1s for any animations
        await new Promise(r => setTimeout(r, 1000));
        const filePath = path.join(SCREENSHOT_DIR, `${p.name}.png`);
        await page.screenshot({ path: filePath, fullPage: true });
        console.log(`Saved: ${filePath}`);
    }

    // For login page, we can log out or just incognito
    console.log('Capturing Login page specifically...');
    const context = await browser.createBrowserContext();
    const page2 = await context.newPage();
    await page2.setViewport({ width: 1280, height: 800 });
    await page2.goto('http://localhost:3000/login', { waitUntil: 'networkidle0' });
    const loginFilePath = path.join(SCREENSHOT_DIR, 'Login.png');
    await page2.screenshot({ path: loginFilePath, fullPage: true });
    console.log(`Saved: ${loginFilePath}`);

    await browser.close();
    console.log('Done!');
}

captureScreenshots().catch(err => {
    console.error('Error capturing screenshots:', err);
    process.exit(1);
});
